src/extension/serializer.ts

```diff
@@ -691,6 +681,60 @@ export class GrpcSerializer extends SerializerBase {
     return Boolean(sessionOutputId)
   }
 
+  // unable to implement marshal methods here, the underlying object structure is the same but
```

Comment:
What are the data types you are referring to that prevents marshal methods from being implemented here?

Reply:
I couldn't pass a type via a type parameter, there are no common base classes, union types are a no go, etc ... which means we have type errors when parsing the code of the functions.  The structure may be the same but the data types nested within differ sometimes. A good example of this is in the method `marshalCellOutputs` where in ts_proto we have `UInt32Value` and `string` whereas es_proto we have `number` and `bigint`.  Aside from all of this there was one anomaly I couldn't explain. In `marshalFrontmatter` the return structure has a `tag` property in ts_proto, but in es_proto it's considered invalid to have it there. I wonder if there was some version of the proto that changed between compiles/transpiles?

Reply:
I think this comment is no longer relevant now that we are going with a single class using the es library.